kevm

```diff
@@ -157,33 +188,48 @@ run_get_failing() {
 # ----
 
 run_command="$1" ; shift
+backend="ocaml"
+params=()
+while [[ $# -gt 0 ]]
+do
+    case "$1" in
+        --backend)
+                backend="$2"
+                shift 2
+                ;;
+        *)
+            params+=("$1")
+            shift
+            ;;
+    esac
+done
+
+set -- "${params[@]}"
+
 case "$run_command" in
 
     # Running
-    run)          run_krun    ocaml   "$@" --interpret ;;
-    run-java)     run_krun    java    "$@"             ;;
-    run-haskell)  run_krun    haskell "$@"             ;;
-    debug)        run_kdebug          "$@"             ;;
-    search)       run_ksearch         "$@"             ;;
-    prove)        run_proof           "$@"             ;;
+    run)          run_krun      "$backend"  "$@"    ;;
+    debug)        run_kdebug                "$@"    ;;
+    search)       run_ksearch               "$@"    ;;
+    prove)        run_proof                 "$@"    ;;
 
     # Testing
-    interpret)    run_interpreter  "$@" ;;
-    test)         run_test         "$@" ;;
-    test-profile) run_test_profile "$@" ;;
-    sort-logs)    run_sort_logs         ;;
-    get-failing)  run_get_failing  "$@" ;;
+    interpret)    run_interpreter               "$@"    ;;
+    test)         run_test         "$backend"   "$@"    ;;
+    test-profile) run_test_profile "$backend"   "$@"    ;;
+    sort-logs)    run_sort_logs                         ;;
+    get-failing)  run_get_failing               "$@"    ;;
 
     *) echo "
     normal usage
     ============
 
-        $0 [run|run-java|run-haskell|debug|search] <pgm>  <K args>*
-        $0 prove                                   <spec> <K args>*
+        $0 run            [--backend <backend>] <pgm>   <K args>*
```

Comment:
Here we even explicitly say that it must come as the first argument after the `run_command`.